### PR TITLE
chore: migrate from ADAL lib before its EOL

### LINKED ISSUE #1635: Migrate from ADAL before its EOL
**Is your feature request related to a problem?/Why is this needed**
According to Microsoft's announcement "ADAL end-of-life is December 31, 2022. While ADAL apps may continue to work, no support or security fixes will be provided past end-of-life. In addition, there are no planned ADAL releases planned prior to end-of-life for features or planned support for new platform versions. We recommend prioritizing migration to Microsoft Authentication Library (MSAL)."

**Describe the solution you'd like in detail**
Use the MSAL library with an adapter so the new authentication can work with current clients (non SDK v2) (see https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/MIGRATION.md).

**Describe alternatives you've considered**
Migrating the clients to the new SDK v2 as detailed in https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/MIGRATION_GUIDE.md. Besides being a lot more work than just using the new auth with an adapter, not all new clients are stable and/or have support to all the functionality existing in their v1 counterparts.

**Additional context**
https://techcommunity.microsoft.com/t5/microsoft-entra-azure-ad-blog/microsoft-entra-change-announcements-september-2022-train/ba-p/2967454

### PR DESCRIPTION
**Pull request by r4f4**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**: migrate from ADAL to MSAL, since ADAL's EOL is approaching.

**Which issue(s) this PR fixes**:

Fixes #1635 

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by r4f4** [comment]
Fixed a typo `s/HasPrefix/HasSuffix/` and renamed commits from `feat:` to `chore:` prefix.

File: test/utils/azure/azure_helpers.go [hunk 1]
```diff
@@ -299,7 +314,16 @@ func getClient(env azure.Environment, subscriptionID, tenantID string, armSpt *a
...
-	authorizer := autorest.NewBearerAuthorizer(armSpt)
+	if !strings.HasSuffix(scope, "/.default") {
+		scope += "/.default"
+	}
+	// Use an adapter so azidentity in the Azure SDK can be used as Authorizer
+	// when calling the Azure Management Packages, which we currently use. Once
+	// the Azure SDK clients (found in /sdk) move to stable, we can update our
+	// clients and they will be able to use the creds directly without the
+	// authorizer.
+	authorizer := azidext.NewTokenCredentialAdapter(cred, []string{scope})
```
Comment (andyzhangx) [inline test/utils/azure/azure_helpers.go:325]: @MartinForReal do you know how to fix this now?

Comment (MartinForReal) [reply test/utils/azure/azure_helpers.go:325]: As mentioned in comments, we need to migrate our model from the ones in /service to the ones in /sdk/resourcemanager

Comment (andyzhangx) [reply test/utils/azure/azure_helpers.go:325]: not an issue now since this folder is testing code.

**Review by andyzhangx — APPROVED** [review]
/hold cancel

**Pull request merged**